package-res/resources/web/vizapi/DataTable.js

```diff
@@ -15,1094 +15,1220 @@
 *
 */
 
-/*
-pentaho.DataTable
-pentaho.DataView
+pentaho = typeof pentaho == "undefined" ? {} : pentaho;
 
-Client-side non-visual data tables.
+/**
+ * Pentaho JavaScript Visualization Data APIs.
+ *
+ * This module defines interfaces and classes
+ * used by visualizations for data representation and manipulatation.
+ *
+ * @module common-ui.vizapi.data
+ * @main common-ui.vizapi.data
+ */
+
+/**
+ * An _AbstractDataTable_ represents a set of tabular data.
+ *
+ * @class AbstractDataTable
+ * @constructor
+ */
+ pentaho.AbstractDataTable = function AbstractDataTable() {};
+
+/**
+ * Gets a column object, given its index.
+ *
+ * @method _getColumn
+ * @param {Number} colIdx The column index (zero-based).
+ * @return {Object} The column object.
+ * @protected
+ */
+
+/**
+ * Gets a row object, given its index.
+ *
+ * @method _getRow
+ * @param {Number} rowIdx The row index (zero-based).
+ * @return {Object} The row object.
+ * @protected
+ */
+
+/**
+ * Gets a cell, given its row and column index.
+ *
+ * @method _getCell
+ * @param {Number} rowIdx The cell's row index (zero-based).
+ * @param {Number} colIdx The cell's column index (zero-based).
+ * @return {Null|Object} The cell.
+ * @protected
+ */
+
+/**
+ * Gets the number of columns.
+ *
+ * @method getNumberOfColumns
+ * @return {Number} The number of columns.
+ */
+
+/**
+ * Gets the number of rows.
+ *
+ * @method getNumberOfRows
+ * @return {Number} The number of rows.
+ */
+
+// TODO: Not publicizing the add method yet, as it is not unit-tested,
+// and would need many other methods to be complete:
+// addRow, addRows, setCell, setValue, setFormattedValue, ...
+// Used internally for the createTrend method.
 
-author: James Dixon
+/*
+ * Appends a new column to the table, given its specification.
+ *
+ * For data views,
+ * this method adds a column to the nearest `DataTable`
+ * and makes the column visible in the intervening views.
+ *
+ * In all existing rows, the value of the new column will be set to `null`.
+ *
+ * @method addColumn
+ * @param {Object} colSpec The column specification.
+ * @param {String} colSpec.id The id of the column.
+ * @param {String} [colSpec.type="string"]  The data type of the column's values.
+ * @param {String} [colSpec.label] The label of the column.
+ * @return {Number} The index of the new column.
+ */
+
+// TODO: metadata/properties on column/cell/row/table should be stored in the metadata bag?
```

Comment:
This comment clarifies, to any future developers, that the current metadata properties implementation is considered wrong. According to Google API, metadata should stored in "p" properties, at the table, column, row and cell levels.
Also, adverts that any change must be made also in Analyzer server-side.
If this gets done in the current VizAPI refactoring, great, the comment is removed. If not, someone else, in the future, will know that this would be a desirable change.

Do you still think I should remove it? Change the text? Change the TAG?